### ISSUE TITLE
"camelcase" should ignore generic arguments like it ignores function arguments

### ISSUE THREAD

**Issue by AlicanC** [gajus/eslint-plugin-flowtype#460]
ESLint [documentation on `camelcase`](https://eslint.org/docs/rules/camelcase) states:

> `"ignoreImports": true` does not check ES2015 imports (but still checks any use of the imports later in the code except function arguments)

I think the "except function arguments" part should also apply to generic arguments.

```ts
import { type PostView_post } from './__generated__/PostView_post.graphql';
import { some_data } from './data';

function MyComponent(props) {
  // PostView_post causes an error and I'm suggesting it shouldn't
  const post = useFragment<PostView_post>(graphql` ... `);

  // some_data does not cause an error as expected
  console.log(some_data);

  return <div>{post.message}</div>
}
```
